spec/registry.md

```diff
@@ -559,6 +559,174 @@ together with the resolved options' values.
 
 The _function_ `:currency` performs selection as described in [Number Selection](#number-selection) below.
 
+### The `:unit` function
+
+The _function_ `:unit` is **Proposed** for inclusion in the next release of this specification but has not yet been finalized.
+The _function_ `:unit` is proposed to be a **RECOMMENDED** selector and formatter for unitized values,
+that is, for numeric values associated with a unit of measurement.
+This is a specialized form of numeric selection and formatting.
+
+#### Operands
+
+The _operand_ of the `:unit` function can be one of any number of
+implementation-defined types,
+each of which contains a numerical `value` plus a `unit`
+or it can be a [Number Operand](#number-operands), as long as the _option_
+`unit` is provided.
+The _option_ `unit` MAY be used to override the units of an implementation-defined type,
+provided the units are compatible and the implementation supports conversion.
```

Comment:
Hmmm. I missed this.

I have never thought of 'unit=x' as involving or requiring conversion, but rather simply using the numeric value of the operand with the specified unit x. I think that you are right; this should take a bit of rewording. Perhaps:

```suggestion
The _option_ `unit` MAY be used to override the unit value of an implementation-defined type,
if the implementation supports unit conversion. 
However, if the units cannot be converted, then the unit value is not overridden and a Bad Option error must be emitted.
```

**Side issue**

Frankly, currency is even worse. Suppose we have input of a implementation-defined parameter $n = {value=300, currency=EUR} and we format or select {$n :currency currency=JPY}. That is really nasty, displaying 300¥. However, there is no stable conversion possible.

So for :currency, we should make it really clear that if there is an implementation-defined operand that contains a currency value, the currency option is invalid, no matter its value.

Reply:
Currency explicitly says that this is an error.

The unit example is a leftover from early on. I will make it use `usage`. We should do the same thing with the `unit` option that we do with the `currency` option: you can use it to create a unit value with a number operand. Otherwise it is an error.

Reply:
Sounds great.